### ISSUE TITLE
Change the reactor temperature to something smiliar but executable

### ISSUE THREAD

**Issue by JensWehner** [coderefinery/testing#179]
For python we replaced the test by this:

```python
# Hint look at monkeypatching for functions
import time
def check_time_to_now_even(seconds):
    """
    Checks whether time difference between now and given time in seconds is even
    """
    if seconds < 0:
        raise ValueError('received negative input')
   
    now=time.time()
    return (seconds-now)%2 == 0
```

with the solution 

```python

#Solution to #4
def test_check_time(monkeypatch):
  def mocktime():
    return 2
  monkeypatch.setattr(time,"time",mocktime)
  assert check_time_to_now_even(0) == 1
  assert check_time_to_now_even(1) == 0

```

**Comment by bast** [comment]
Good suggestion to turn it to something executable. Thanks!

Similar idea could be to test the following function without waiting for 4 years:
```python
import datetime

def we_are_in_a_leap_year():
    year = datetime.datetime.now().year
    try:
        # function throws ValueError if the date (Feb 29) does not exist for
        # this year
        datetime.datetime(year, 2, 29)
        return True
    except ValueError:
        return False
```

**Comment by JensWehner** [comment]
Yes in the end I do not care, but most students actually wrote tests for this exercise, which was really nice to see how engaged they were. The reactor exercise confused many and we spend a lot of time explaining that this code does not work. So I think the executable code is much better, which exact one you take I do not care. :) Yours is maybe already more relatable and less construed.

**Comment by bast** [comment]
I agree the reactor one is weird and also full support to replace it.

**Comment by JensWehner** [comment]
So I make a PR and we merge it or is there any reason that would hold this up?

**Comment by bast** [comment]
Yes please go ahead with the PR. Thanks a lot for this suggestion. When I taught it, I always just skimmed over it so you have more experience here what works well and what doesn't.

**Comment by rantahar** [comment]
Am I right in thinking some languages are still missing? I will not merge yet, but we should get back to this.

**Comment by JensWehner** [comment]
The problem with other languages is, that it it requires a lot of boilerplate to make the test work. Or even Macros, dependency injection or other stuff.

**Comment by mmesiti** [comment]
I think that this "nightmare scenario" is kind of interesting, especially with the other languages where you can't cheat with monkeypatching and need to talk about injection points (which also links to the question "how do you test functions higher in the function hierarchy without doing expensive integration or e2e tests?", which I think everyone writing tests is bound to face)

**Comment by JensWehner** [comment]
Today, I would argue not to promote monkey patching and instead promote dependency injection instead

**Comment by mmesiti** [comment]
So one could use the same approach for all programming languages (architecture change + dependency injection) and say that dynamic languages also allow monkeypatching?
